src/pudl/transform/ferc714.py

```diff
@@ -293,6 +323,36 @@ def _assign_respondent_id_ferc714(
     return df
 
 
+def _fillna_respondent_id_ferc714_source(
+    df: pd.DataFrame, source: Literal["csv", "xbrl"]
+) -> pd.DataFrame:
+    """Fill missing CSV or XBRL respondent id.
+
+    The source (CSV or XBRL) tables get assigned a PUDL-derived
+    ``respondent_id_ferc714`` ID column (via :func:`_assign_respondent_id_ferc714`).
+    After we concatenate the source tables, we sometimes to backfill and
+    forward-fill the source IDs (``respondent_id_ferc714_csv`` and
+    ``respondent_id_ferc714_xbrl``). This way the older records from the CSV years
+    will also have the XBRL ID's and vice versa. This will enable users to find
+    the full timeseries of a respondent that given either source ID (instead of
+    using the source ID to find the PUDL-derived ID and then finding the records).
```

Comment:
ohh ok I understand now! It's backfilling old, non XBRL years and forward filling non CSV years.

There's one little type here: `we sometimes to backfill`